lua/packer.lua

```diff
@@ -178,8 +180,17 @@ packer.install = function(...)
       await(a.main)
       plugin_utils.update_helptags(install_paths)
       plugin_utils.update_rplugins()
-      local delta = string.gsub(vim.fn.reltimestr(vim.fn.reltime(start_time)), ' ', '')
-      display_win:final_results(results, delta)
+
+      if not vim.tbl_isempty(results.rocks) then
+        await(neorocks.install_rocks(results.rocks))
```

Comment:
Is there a reason (efficiency, etc.) to wait until after all the plugin installs to do the Luarocks installs? If not, I think I would prefer to add the Luarocks install step inside the installer (like the update hooks), for higher throughput. I could make this change after merging the initial PR, if you'd prefer.

As a nitpick here, we should probably handle rock installs before updating rplugins.

Reply:
I don't think we should do it inside of each plugin because there is a chance:

1. they try to install to the same place
2. they try to install conflicting deps
3. they just end up getting messed up because luarocks wasn't meant to be concurrent.

So I have them running sequentially at the end right now because that's the only way I could think of solving these problems. Perhaps there is a way to fix it though.

Reply:
Hmm, fair points (particularly (3)). (2) seems like it could be an issue regardless of how we run luarocks - do you know how luarocks handles conflicting/multiple versions of packages in "normal" use? If it doesn't already handle that for us, we might need to install each plugin's rocks to a separate location. That adds some complication/increases disk space use by duplicating packages, but does make things like rock removal, etc. much easier as well. If we do need to use that approach, then I still think running the install inside each plugin is better; if we can avoid that approach, async-running the install after all the plugin installs should be fine.

Reply:
So I thinkg lua rocks itself already handles the fact that when you remove one package it cleans up the deps of the package (if no one else is using it). But I haven't followed up on that. We can always add that as an enhancement later (since I think our implementation may get a little more complicated over time :laughing: )